R/merge.R

```diff
@@ -158,32 +163,46 @@ setGeneric("mergeSamples",
 
 #' @importFrom S4Vectors SimpleList
 #' @importFrom scuttle sumCountsAcrossFeatures
-.merge_rows <- function(x, f, archetype = 1L, 
+.merge_rows <- function(x, f, archetype = 1L,
+                        na.rm = FALSE,
                         average = FALSE,
                         BPPARAM = SerialParam(),
                         check.assays = TRUE,
                         ...){
     # input check
+    if(!.is_a_bool(na.rm)){
+        stop("'na.rm' must be TRUE or FALSE.", call. = FALSE)
+    }
     if( !.is_a_bool(average) ){
         stop("'average' must be TRUE or FALSE.", call. = FALSE)
     }
     if( !.is_a_bool(check.assays) ){
         stop("'check.assays' must be TRUE or FALSE.", call. = FALSE)
     }
     #
+    # Make a vector from the taxonomic data.
+    rank <- taxonomyRanks(x)[1]
+    col <- which( taxonomyRanks(x) %in% rank )
+    tax_cols <- .get_tax_cols_from_se(x)
+
+    # if na.rm is TRUE, convert NA values to "NA" in f.
+    if (na.rm) {
+        f[is.na(f)] <- "NA"
+    }
```

Comment:
Use similar coding style than in other parts of the function

`if( na.rm )`

Also add comment, why this is done. --> This is done because otherwise we lose NA values.

It is not clear from the code what is the intention

Reply:
ACK.